.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,216 @@
+name: Pull Request CI
+run-name: ${{ github.actor }} is running Pull Request CI
+
+on:
+  pull_request_target:
+    types: [opened, labeled, reopened, synchronize]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  initialize:
+    runs-on: ubuntu-latest
+    env:
+      AWS_BUCKET: code.s4d.io
+      AWS_REGION: us-east-1
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.13.1
+
+      - name: Clean yarn cache
+        run: yarn cache clean
+
+      - name: Initialize dependencies
+        run: yarn install
+
+      - name: Cache Dependencies
+        uses: actions/cache@v3
+        if: steps.validate-dependencies.outputs.cache-hit != 'true'
+        with:
+          path: "**/node_modules"
+          key: node-modules-${{ hashFiles('./yarn.lock') }}
+
+  unit_tests_and_linting:
+    runs-on: ubuntu-latest
+    needs: initialize
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.13.1
+
+      - name: Uncache Dependencies
```

Comment:
Small doubt regarding the label name here, are we not caching the node modules here?

Reply:
Caching is happening at line 34.